engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -649,7 +649,7 @@ protected void serviceStart() throws Exception
       addSSLConfigResource(config);
       WebApp webApp = WebApps.$for("stram", StramAppContext.class, appContext, "ws").with(config).start(new StramWebApp(this.dnmgr));
       LOG.info("Started web service at port: " + webApp.port());
-      appMasterTrackingUrl = NetUtils.getConnectAddress(webApp.getListenerAddress()).getHostName() + ":" + webApp.port();
+      appMasterTrackingUrl = NetUtils.getConnectAddress(webApp.getListenerAddress()).getAddress().getCanonicalHostName() + ":" + webApp.port();
```

Comment:
It is not possible to reliably determine FQDN or even IP address on a server itself that another host may use to communicate with the server. We should only cover 80% of standard configurations. For remaining 20% it may not work and can be covered by an explicit overwrite.

@sanjaypujare my machine is not part of any domain and DNS server does not have records for it. Likely the same will apply to sandbox/docker container environment as well.